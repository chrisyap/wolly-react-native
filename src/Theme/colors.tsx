import { shade, tint, readableColor, saturate } from 'polished';

const getShades = (color: string, steps: number, dark?: boolean): Array<string> => {
  let lights = [];
  const darks = [];
  let lightArrays = [];
  let darkArrays = [];
  for (let i = 1; i <= steps; i += 1) {
    const col = tint(i * (1 / (steps + 1)), color);
    lights.push(col);
  }
  for (let i = 1; i <= steps; i += 1) {
    const col = shade(i * (1 / (steps + 1)), color);
    darks.push(col);
  }
  lights = lights.reverse();
  lightArrays = [...lights, color].concat(darks);
  darkArrays = [...lightArrays].reverse();

  if (dark) {
    return darkArrays;
  }
  return lightArrays;
};

const reverseArray = (array: Array<string>): Array<string> => {
  const newArray = [...array].reverse();
  return newArray;
};

export interface ColorsProps {
  text: string;
  black: string;
  blacks: Array<string>;
  white: string;

  primary: string;
  primaries: Array<string>;
  primaryText: string;

  secondary: string;
  secondaries: Array<string>;
  secondaryAccent: string;
  secondaryAccentText: string;

  red: string;
  reds: Array<string>;

  bluegrey: string;
  bluegreys: Array<string>;

  sea: string;
  seas: Array<string>;

  blueberry: string;
  blueberries: Array<string>;

  sky: string;
  skies: Array<string>;

  lime: string;
  limes: Array<string>;

  tangerine: string;
  tangerines: Array<string>;

  success: string;
  successes: Array<string>;
  successText: string;

  info: string;
  infos: Array<string>;
  infoText: string;

  warning: string;
  warnings: Array<string>;
  warningText: string;

  danger: string;
  dangers: Array<string>;
  dangerText: string;

  texts: Array<string>;
  backgrounds: Array<string>;
  borders: Array<string>;

  modalBackdrop: string;
  isDarkTheme: boolean;
}

const primaryColors = {
  red: '#c20000',
  blueGrey: '#4c626c',
  black: '#000000',
};

const secondaryColors = {
  sea: '#152773',
  blueberry: '#572381',
  sky: '#005aa3',
  lime: '#a4b123',
  tangerine: '#e9600e',
};
const contextual = {
  success: '#037d04',
  info: '#005aa3',
  warning: '#F36E20',
  danger: '#d20000',
};

const secondaryDark = {
  secondary: saturate(0.1, tint(0.4, primaryColors.blueGrey)),
  secondaryAccent: saturate(0.15, primaryColors.blueGrey),
};

const contextualDark = {
  success: saturate(0.3, tint(0.2, contextual.success)),
  info: saturate(0.4, tint(0.38, contextual.info)),
  warning: saturate(0.3, tint(0.2, contextual.warning)),
  danger: saturate(0.3, tint(0.47, contextual.danger)),
};

const colors = {
  text: '#111111',

  black: primaryColors.black,
  blacks: getShades(primaryColors.black, 10),
  white: '#ffffff',

  primary: primaryColors.red,
  primaries: getShades(primaryColors.red, 10),
  primaryText: readableColor(primaryColors.red, '#000000', '#ffffff'),
  secondary: primaryColors.blueGrey,
  secondaries: getShades(primaryColors.blueGrey, 10),
  secondaryAccent: primaryColors.blueGrey,
  secondaryAccentText: readableColor(primaryColors.blueGrey, '#000000', '#ffffff'),

  red: primaryColors.red,
  reds: getShades(primaryColors.red, 10),

  bluegrey: primaryColors.blueGrey,
  bluegreys: getShades(primaryColors.blueGrey, 10),
  sea: secondaryColors.sea,
  seas: getShades(secondaryColors.sea, 10),

  blueberry: secondaryColors.blueberry,
  blueberries: getShades(secondaryColors.blueberry, 10),

  sky: secondaryColors.sky,
  skies: getShades(secondaryColors.sky, 10),

  lime: secondaryColors.lime,
  limes: getShades(secondaryColors.lime, 10),

  tangerine: secondaryColors.tangerine,
  tangerines: getShades(secondaryColors.tangerine, 10),

  success: contextual.success,
  successes: getShades(contextual.success, 10),
  successText: readableColor(contextual.success, '#000000', '#ffffff'),

  info: contextual.info,
  infos: getShades(contextual.info, 10),
  infoText: readableColor(contextual.info, '#000000', '#ffffff'),

  warning: contextual.warning,
  warnings: getShades(contextual.warning, 10),
  warningText: readableColor(contextual.warning, '#000000', '#ffffff'),

  danger: contextual.danger,
  dangers: getShades(contextual.danger, 10),
  dangerText: readableColor(contextual.danger, '#000000', '#ffffff'),
};

const lightThemeColors = {
  ...colors,
  texts: [colors.text, colors.secondary, colors.secondaries[5]],
  backgrounds: [colors.secondaries[1], colors.white, colors.blacks[4]],
  borders: [colors.secondaries[2], colors.blacks[3]],
  modalBackdrop: primaryColors.black,
  btnGroupBorder: primaryColors.blueGrey,
  rowOdd: tint(0.5, colors.secondaries[0]),
  isDarkTheme: false,
};

const colorsDark = {
  text: '#eeeeee',

  black: '#ffffff',
  blacks: reverseArray(getShades(colors.black, 10, true)),
  white: primaryColors.black,

  primary: primaryColors.red,
  primaries: reverseArray(colors.primaries),
  primaryText: readableColor(primaryColors.red, '#000000', '#ffffff'),
  secondary: secondaryDark.secondary,
  secondaries: getShades(secondaryDark.secondary, 10, true),

  secondaryAccent: secondaryDark.secondaryAccent,
  secondaryAccentText: readableColor(secondaryDark.secondaryAccent, '#000000', '#ffffff'),

  red: primaryColors.red,
  reds: reverseArray(colors.reds),

  bluegrey: primaryColors.blueGrey,
  bluegreys: reverseArray(colors.bluegreys),
  sea: secondaryColors.sea,
  seas: reverseArray(colors.seas),

  blueberry: secondaryColors.blueberry,
  blueberries: reverseArray(colors.blueberries),

  sky: secondaryColors.sky,
  skies: reverseArray(colors.skies),

  lime: secondaryColors.lime,
  limes: reverseArray(colors.limes),

  tangerine: secondaryColors.tangerine,
  tangerines: reverseArray(colors.tangerines),

  success: contextualDark.success,
  successes: getShades(contextualDark.success, 10, true),
  successText: readableColor(contextualDark.success, '#000000', '#ffffff'),

  info: contextualDark.info,
  infos: getShades(contextualDark.info, 10, true),
  infoText: readableColor(contextualDark.info, '#000000', '#ffffff'),

  warning: contextualDark.warning,
  warnings: getShades(contextualDark.warning, 10, true),
  warningText: readableColor(contextualDark.warning, '#000000', '#ffffff'),

  danger: contextualDark.danger,
  dangers: getShades(contextualDark.danger, 10, true),
  dangerText: readableColor(contextualDark.danger, '#000000', '#ffffff'),

  texts: [colors.white, colors.secondaries[4], colors.secondaries[8]],
  backgrounds: [colors.secondaries[8], colors.black],
  borders: [colors.secondaries[6], colors.blacks[7]],
};

const darkThemeColors = {
  ...colorsDark,
  modalBackdrop: secondaryDark.secondaryAccent,
  btnGroupBorder: secondaryDark.secondaryAccent,
  rowOdd: colorsDark.secondaries[1],
  isDarkTheme: true,
};

const updateBackground = (): void => {
  colorsDark.secondaries.splice(
    0,
    1,
    tint(
      0.02,
      colorsDark.secondaries[0]
      // 'pink'
    )
  );
};

updateBackground();

export { colors, darkThemeColors, lightThemeColors };

// console.log('dark', JSON.stringify(darkThemeColors, null, 4));
