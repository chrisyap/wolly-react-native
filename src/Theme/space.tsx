// const space = [0, 2, 4, 8, 16, 32, 64, 128, 256, 512];
const space: Array<string> = [];
const spacer = 16;

for (let i = 0; i < 20; i += 1) {
  space.push(`${spacer * (i * 0.5)}px`);
}

export { space, spacer };
