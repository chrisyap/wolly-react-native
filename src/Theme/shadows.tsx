import { colors } from './colors';
import { rgba } from 'polished';

const shadows: Array<string> = [];
const steps = 25;

for (let i = 0; i < steps; i += 1) {
  const sha = `
  0 ${steps * i * 0.03}px ${steps * i * 0.05}px 0 ${rgba(colors.bluegrey, 0.1)}, 
  0 ${steps * i * 0.02}px ${steps * i * 0.1}px 0 ${rgba(colors.bluegrey, 0.1)}`;
  shadows.push(sha);
}

export { shadows };
