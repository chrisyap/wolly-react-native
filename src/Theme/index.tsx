import 'styled-components/native';
// import { DefaultTheme } from 'styled-components/native';

import { colors, darkThemeColors, lightThemeColors, ColorsProps } from './colors';
import { space, spacer } from './space';
import { shadows } from './shadows';

export const fontSizes = [
  '8px', // 0
  '9px', // 1
  '10px', // 2
  '11px', // 3
  '12px', // 4
  '13px', // 5
  '14px', // 6
  '16px', // 7
  '18px', // 8
  '20px', // 9
  '24px', // 10
  '26px', // 11
  '28px', // 12
  '32px', // 13
  '34px', // 14
  '41px', // 15
  '46px', // 16
  '52px', // 17
  '58px', // 18
  '66px', // 19
  '74px', // 20
  '83px', // 21
  '94px', // 22
  '105px', // 23
];

// export const regular = 400;
// export const bold = 700;

// styled-system's `fontWeight` function can hook into the `fontWeights` object
export const fontWeights = {
  light: 300,
  normal: 400,
  medium: 500,
  semibold: 600,
  bold: 700,
};

export const lineHeights = {
  standard: '22px',
  display: '20px',
};

// const letterSpacings = {
//   normal: 'normal',
//   caps: '0.025em',
// };

// export const textStyles = {
//   display8: {
//     fontSize: fontSizes[8] + 'px',
//     fontWeight: fontWeights.bold,
//     lineHeight: lineHeights.display,
//   },
//   display7: {
//     fontSize: fontSizes[7] + 'px',
//     fontWeight: fontWeights.bold,
//     lineHeight: lineHeights.display,
//   },
//   display6: {
//     fontSize: fontSizes[6] + 'px',
//     fontWeight: fontWeights.bold,
//     lineHeight: lineHeights.display,
//   },
//   display5: {
//     fontSize: fontSizes[5] + 'px',
//     fontWeight: fontWeights.bold,
//     lineHeight: lineHeights.display,
//   },
//   display4: {
//     fontSize: fontSizes[4] + 'px',
//     fontWeight: fontWeights.bold,
//     lineHeight: lineHeights.display,
//   },
//   display3: {
//     fontSize: fontSizes[3] + 'px',
//     fontWeight: fontWeights.bold,
//     lineHeight: lineHeights.display,
//   },
//   display2: {
//     fontSize: fontSizes[2] + 'px',
//     fontWeight: fontWeights.bold,
//     lineHeight: lineHeights.display,
//   },
//   display1: {
//     fontSize: fontSizes[1] + 'px',
//     fontWeight: fontWeights.bold,
//     lineHeight: lineHeights.display,
//   },
//   display0: {
//     fontSize: fontSizes[0] + 'px',
//     fontWeight: fontWeights.bold,
//     lineHeight: lineHeights.display,
//     letterSpacing: letterSpacings.caps,
//     textTransform: 'uppercase',
//   },
//   body2: {
//     fontSize: fontSizes[2] + 'px',
//     fontWeight: fontWeights.normal,
//     lineHeight: lineHeights.standard,
//   },
//   body1: {
//     fontSize: fontSizes[1] + 'px',
//     fontWeight: fontWeights.normal,
//     lineHeight: lineHeights.standard,
//   },
//   body0: {
//     fontSize: fontSizes[0] + 'px',
//     fontWeight: fontWeights.normal,
//     lineHeight: lineHeights.standard,
//   },
// };

// export const colorStyles = {
//   whiteOnText: {
//     color: colors.white,
//     backgroundColor: colors.text,
//   },
//   whiteOnGray: {
//     color: colors.white,
//     backgroundColor: colors.gray,
//   },
//   textOnLightGray: {
//     color: colors.text,
//     backgroundColor: colors.lightGray,
//   },
//   whiteOnBlue: {
//     color: colors.white,
//     backgroundColor: colors.blue,
//   },
//   blueOnLightBlue: {
//     color: colors.blue,
//     backgroundColor: colors.lightBlue,
//   },
//   whiteOnGreen: {
//     color: colors.white,
//     backgroundColor: colors.green,
//   },
//   greenOnLightGreen: {
//     color: colors.green,
//     backgroundColor: colors.lightGreen,
//   },
//   whiteOnRed: {
//     color: colors.white,
//     backgroundColor: colors.red,
//   },
//   redOnLightRed: {
//     color: colors.red,
//     backgroundColor: colors.lightRed,
//   },
//   textOnOrange: {
//     color: colors.text,
//     backgroundColor: colors.orange,
//   },
//   whiteOnPurple: {
//     color: colors.white,
//     backgroundColor: colors.purple,
//   },
//   purpleOnLightPurple: {
//     color: colors.purple,
//     backgroundColor: colors.lightPurple,
//   },
//   textOnWhite: {
//     color: colors.text,
//     backgroundColor: colors.white,
//   },
//   grayOnWhite: {
//     color: colors.gray,
//     backgroundColor: colors.white,
//   },
//   blueOnWhite: {
//     color: colors.blue,
//     backgroundColor: colors.white,
//   },
//   greenOnWhite: {
//     color: colors.green,
//     backgroundColor: colors.white,
//   },
//   redOnWhite: {
//     color: colors.red,
//     backgroundColor: colors.white,
//   },
//   purpleOnWhite: {
//     color: colors.purple,
//     backgroundColor: colors.white,
//   },
//   whiteOnDarkOrange: {
//     color: colors.white,
//     backgroundColor: colors.darkOrange,
//   },
// };

// colorStyles.info = colorStyles.textOnLightGray;
// colorStyles.success = colorStyles.whiteOnGreen;
// colorStyles.warning = colorStyles.textOnOrange;
// colorStyles.danger = colorStyles.whiteOnRed;

// styled-system's `borderRadius` function can hook into the `radii` object/array
export const radii = ['0px', '2px', '4px', '8px', '16px'];
export const radius = '2px';
export const radiusRounded = '290486px';

// animation duration
export const duration = {
  fast: `150ms`,
  normal: `300ms`,
  slow: `450ms`,
  slowest: `600ms`,
};

// animation easing curves
const easeInOut = 'cubic-bezier(0.5, 0, 0.25, 1)';
const easeOut = 'cubic-bezier(0, 0, 0.25, 1)';
const easeIn = 'cubic-bezier(0.5, 0, 1, 1)';

const timingFunctions = {
  easeInOut,
  easeOut,
  easeIn,
};

// animation delay
const transitionDelays = {
  small: `60ms`,
  medium: `160ms`,
  large: `260ms`,
  xLarge: `360ms`,
};

export interface ThemeProps {
  space: Array<string>;
  spacer: number;
  fontSizes: Array<string>;
  fontWeights: {
    light: number;
    normal: number;
    medium: number;
    semibold: number;
    bold: number;
  };
  lineHeights: {
    standard: string;
    display: string;
  };
  colors: ColorsProps;
  radii: Array<string>;
  radius: string;
  radiusRounded: string;
  shadows: Array<string>;
  duration: { fast: string; normal: string; slow: string; slowest: string };
  timingFunctions: { easeInOut: string; easeOut: string; easeIn: string };
  transitionDelays: {
    small: string;
    medium: string;
    large: string;
    xLarge: string;
  };
}

const theme = {
  space,
  spacer,
  // font,
  fontSizes,
  fontWeights,
  lineHeights,
  // letterSpacings,
  // regular,
  // bold,
  // textStyles,
  // colors,
  // colorStyles,
  radii,
  radius,
  radiusRounded,
  shadows,
  // maxContainerWidth,
  duration,
  timingFunctions,
  transitionDelays,
};

export default { ...theme, colors };

export const darkTheme = {
  ...theme,
  colors: darkThemeColors,
};

export const lightTheme = {
  ...theme,
  colors: lightThemeColors,
};

// console.log('dark', darkThemeColors, 'light', lightThemeColors);
