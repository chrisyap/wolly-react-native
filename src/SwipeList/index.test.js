import React from "react";
import List from "./index";
import { ThemeConsumer } from "styled-components";
import { darkTheme, lightTheme } from "../Theme";

const shallowWithTheme = (children, theme = lightTheme) => {
  ThemeConsumer._currentValue = theme;
  return shallow(children);
};

describe("List", () => {
  it("renders correctly", () => {
    const wrapper = shallowWithTheme(<List />, lightTheme);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders dark theme correctly", () => {
    const wrapper = shallowWithTheme(<List />, darkTheme);
    expect(wrapper).toMatchSnapshot();
  });
});
