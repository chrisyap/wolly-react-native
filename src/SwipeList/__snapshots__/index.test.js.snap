// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders correctly 1`] = `
<Styled(View)
  flexGrow={1}
  flexShrink={1}
  position="relative"
>
  <SwipeListView
    closeOnRowBeginSwipe={false}
    closeOnRowOpen={true}
    closeOnRowPress={true}
    closeOnScroll={true}
    directionalDistanceChangeThreshold={2}
    disableHiddenLayoutCalculation={false}
    disableLeftSwipe={false}
    disableRightSwipe={false}
    leftOpenValue={0}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    previewFirstRow={false}
    previewRepeat={false}
    previewRepeatDelay={1000}
    recalculateHiddenLayout={false}
    rightOpenValue={0}
    scrollEventThrottle={16}
    showsVerticalScrollIndicator={false}
    style={
      Object {
        "flexGrow": 1,
      }
    }
    swipeToClosePercent={50}
    swipeToOpenPercent={50}
    swipeToOpenVelocityContribution={0}
    useAnimatedList={false}
    useNativeDriver={true}
  />
  <Scrollbar
    height={0}
    opacity={0}
    positionRight="2px"
    positionScroll={0}
    width={3}
  />
</Styled(View)>
`;

exports[`List renders dark theme correctly 1`] = `
<Styled(View)
  flexGrow={1}
  flexShrink={1}
  position="relative"
>
  <SwipeListView
    closeOnRowBeginSwipe={false}
    closeOnRowOpen={true}
    closeOnRowPress={true}
    closeOnScroll={true}
    directionalDistanceChangeThreshold={2}
    disableHiddenLayoutCalculation={false}
    disableLeftSwipe={false}
    disableRightSwipe={false}
    leftOpenValue={0}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    previewFirstRow={false}
    previewRepeat={false}
    previewRepeatDelay={1000}
    recalculateHiddenLayout={false}
    rightOpenValue={0}
    scrollEventThrottle={16}
    showsVerticalScrollIndicator={false}
    style={
      Object {
        "flexGrow": 1,
      }
    }
    swipeToClosePercent={50}
    swipeToOpenPercent={50}
    swipeToOpenVelocityContribution={0}
    useAnimatedList={false}
    useNativeDriver={true}
  />
  <Scrollbar
    height={0}
    opacity={0}
    positionRight="2px"
    positionScroll={0}
    width={3}
  />
</Styled(View)>
`;
